examples/ogbn_train.py

```diff
@@ -0,0 +1,321 @@
+parser.add_argument('--wd', type=float, default=0.0)
+parser.add_argument('--dropout', type=float, default=0.5)
+parser.add_argument(
+    '--use_directed_graph',
+    action='store_true',
+    help='Whether or not to use directed graph',
+)
+args = parser.parse_args()
+device = torch.device(args.device)
+num_epochs = args.epochs
+num_layers = args.num_layers
+num_workers = args.num_workers
+num_hidden_channels = args.hidden_channels
+batch_size = args.batch_size
+neighbors = args.neighbors.split(',')
+num_neighbors = [int(i) for i in neighbors]
+log_interval = args.log_interval
+
+root = osp.join(args.dataset_dir, args.dataset)
+print('The root is: ', root)
+dataset = PygNodePropPredDataset(name=args.dataset, root=root)
+split_idx = dataset.get_idx_split()
+evaluator = Evaluator(name=args.dataset)
+
+data = dataset[0]
+if not args.use_directed_graph:
+    data.edge_index = to_undirected(data.edge_index, reduce='mean')
+
+data.to(device, 'x', 'y')
+
+train_loader = NeighborLoader(
+    data,
+    input_nodes=split_idx['train'],
+    num_neighbors=[args.fan_out] * num_layers,
+    batch_size=batch_size,
+    shuffle=True,
+    num_workers=num_workers,
+    persistent_workers=True,
+)
+val_loader = NeighborLoader(
+    data,
+    input_nodes=split_idx['valid'],
+    num_neighbors=[args.fan_out] * num_layers,
+    batch_size=batch_size,
+    shuffle=True,
+    num_workers=num_workers,
+    persistent_workers=True,
+)
+test_loader = NeighborLoader(
+    data,
+    input_nodes=split_idx['test'],
+    num_neighbors=[args.fan_out] * num_layers,
+    batch_size=batch_size,
+    shuffle=True,
+    num_workers=num_workers,
+    persistent_workers=True,
+)
+subgraph_loader = NeighborLoader(
+    data,
+    input_nodes=None,
+    num_neighbors=[-1],
+    batch_size=batch_size,
+    num_workers=num_workers,
+    persistent_workers=True,
+)
+
+
+class SAGE(torch.nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        hidden_channels: int,
+        out_channels: int,
+        num_layers: int,
+        dropout: float,
+    ) -> None:
+        super().__init__()
+        self.num_layers = num_layers
+        self.convs = torch.nn.ModuleList()
+        self.convs.append(SAGEConv(in_channels, hidden_channels))
+        for _ in range(num_layers - 2):
+            self.convs.append(SAGEConv(hidden_channels, hidden_channels))
+        self.convs.append(SAGEConv(hidden_channels, out_channels))
+        self.dropout = dropout
+
+    def reset_parameters(self) -> None:
+        for conv in self.convs:
+            conv.reset_parameters()
+
+    def forward(self, x: Tensor, edge_index: Tensor) -> Tensor:
+        for i, conv in enumerate(self.convs):
+            x = conv(x, edge_index)
+            if i != self.num_layers - 1:
+                x = x.relu()
+                x = F.dropout(x, p=self.dropout, training=self.training)
+        return x
+
+    def inference(self, x_all: Tensor) -> Tensor:
```

Comment:
It'd be nice to run this method within the script

Reply:
Will add the code to test inference method